lib/system/shared/mesh-client.js

```diff
@@ -221,7 +221,17 @@
 
     var client = {
       _mesh: {},
-      log: opts.log
+      log: opts.log,
+      updateSession:function(session){
+        var _this = this;
+        _this.session = session;
+        if (_this._mesh && _this._mesh.endpoints){
+          Object.keys(_this._mesh.endpoints).forEach(function(endpointName){
+              var endpoint = _this._mesh.endpoints[endpointName];
+              endpoint.messenger.session.id = session.id;
```

Comment:
also - for browser clients, there is only 1 endpoint, but the client code is re-used for the mesh to mesh comms, thus we need to iterate through endpoints and update their messengers, an improvement could be to make the session shared across endpoints by reference - removing the complexity of an update when it changes at the client/origin.